code/game/objects/structures/cannon.dm

```diff
@@ -0,0 +1,104 @@
+/obj/structure/cannon
+	name = "cannon"
+	desc = "Holemaker Deluxe: A sporty model with a good stop power. Any cannon enthusiast should be expected to start here."
+	density = TRUE
+	anchored = TRUE
+	icon_state = "falconet_patina"
+	max_integrity = 300
+	var/obj/item/stack/cannonball/loaded_cannonball = null
+	var/charge_ignited = FALSE
+	var/fire_delay = 15
+	var/charge_size = 15
+	var/fire_sound = 'sound/weapons/gun/general/cannon.ogg'
+
+/obj/structure/cannon/Initialize()
+	. = ..()
+	create_reagents(charge_size)
+
+/obj/structure/cannon/proc/fire()
+	for(var/mob/M in urange(10, src))
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names